src/artboard.cpp

```diff
@@ -395,7 +395,7 @@ bool Artboard::updateComponents()
 				// If the update changed the dirt depth by adding dirt to
 				// something before us (in the DAG), early out and re-run
 				// the update.
-				if (m_DirtDepth < i)
+				if (m_DirtDepth < (unsigned int) i)
```

Comment:
One minor nitpick here, I think we can just change the definition for i in the for loop to unsigned int:
```
for (unsigned int i = 0; i < count; i++)
```
No warnings in clang for that, let me know if that causes issues in gcc...

Reply:
![image](https://user-images.githubusercontent.com/454182/113806015-6b49b780-9716-11eb-9bff-8c2f9b7c9dc1.png)

Reply:
@luigi-rosso that case.. still have a warning due to "count" type.

../submodule/src/artboard.cpp:383:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
    for (unsigned int i = 0; i < count; i++)

Reply:
Should be able to make count an unsigned int too...

Reply:
@luigi-rosso updated. if it's acceptable, i think you can use squash to sum up to one commit (instead of dirty 3 commits)

Otherwise, please ping me.